test/sqllogictest/transform/normalize_lets.slt

```diff
@@ -63,6 +63,45 @@ Explained Query:
 
 EOF
 
+## Test inlining at an inner nesting level. (#18889)
+## `bar` is used only in `baz`, so it should be inlined. The inner WMR should have only one cte.
+query T multiline
+EXPLAIN WITH MUTUALLY RECURSIVE
```

Comment:
Do we also currently cover the case where we have a non-recurisve (`Let`) CTE at the root or non-recursive CTEs between the recrusive (`LetRec`) CTEs?

```sql
WITH 
  cte0 AS (
    -- non-recursive CTE
  )
  ( 
    WITH MUTUALLY RECURSIVE
    -- possibly with further non-recursive CTEs
  )

Reply:
Yes, we cover the case where the situation that you described occurs just before the entire `NormalizeLets`. I'll add a test for this.

But note that at the point where `inline_lets` is called, we don't have any `Let` nodes anywhere in the tree. (I might add an assert for this.) At this point, all of them have been normalized to be `LetRec` nodes. (`promote_let_rec` does this through `digest_lets_helper`.) And then if there is no actual recursion, then they get turned back into a stack of `Let`s at the end of `action` at the `// Disassemble `LetRec` into a `Let` stack if possible.`